internal/impl/azure/input_queue_storage.go

```diff
@@ -119,60 +121,60 @@ func (a *azureQueueStorage) ReadBatch(ctx context.Context) (batch service.Messag
 		err = fmt.Errorf("queue name interpolation error: %w", err)
 		return
 	}
-	queueURL := a.conf.svcURL.NewQueueURL(queueName)
-	messageURL := queueURL.NewMessagesURL()
+	queueClient := a.conf.client.NewQueueClient(queueName)
 	var approxMsgCount int32
 	if a.conf.TrackProperties {
-		if props, err := queueURL.GetProperties(ctx); err == nil {
-			approxMsgCount = props.ApproximateMessagesCount()
+		if props, err := queueClient.GetProperties(ctx, nil); err == nil {
+			if amc := props.ApproximateMessagesCount; amc != nil {
+				approxMsgCount = *amc
+			}
 		}
 	}
-	dequeue, err := messageURL.Dequeue(ctx, int32(a.conf.MaxInFlight), a.conf.DequeueVisibilityTimeout)
+	visibilityTimeout := int32(a.conf.DequeueVisibilityTimeout.Seconds())
+	numMessages := int32(a.conf.MaxInFlight)
+	dequeue, err := queueClient.DequeueMessages(ctx, &azq.DequeueMessagesOptions{
+		NumberOfMessages:  &numMessages,
+		VisibilityTimeout: &visibilityTimeout,
+	})
 	if err != nil {
-		if cerr, ok := err.(azqueue.StorageError); ok {
-			if cerr.ServiceCode() == azqueue.ServiceCodeQueueNotFound {
-				ctx := context.Background()
-				_, err = queueURL.Create(ctx, azqueue.Metadata{})
+		if cerr, ok := err.(*azcore.ResponseError); ok {
+			if cerr.StatusCode == http.StatusNotFound {
+				_, err = queueClient.Create(ctx, nil)
 				return nil, nil, err
 			}
 			return nil, nil, fmt.Errorf("storage error message: %v", cerr)
 		}
 		return nil, nil, fmt.Errorf("error dequeing message: %v", err)
 	}
-	if n := dequeue.NumMessages(); n > 0 {
-		props, _ := queueURL.GetProperties(ctx)
-		metadata := props.NewMetadata()
-		dqm := make([]*azqueue.DequeuedMessage, n)
-		for i := int32(0); i < n; i++ {
-			queueMsg := dequeue.Message(i)
-			part := service.NewMessage([]byte(queueMsg.Text))
-			part.MetaSetMut("queue_storage_insertion_time", queueMsg.InsertionTime.Format(time.RFC3339))
-			part.MetaSetMut("queue_storage_queue_name", queueName)
-			if a.conf.TrackProperties {
-				msgLag := 0
-				if approxMsgCount >= n {
-					msgLag = int(approxMsgCount - n)
-				}
-				part.MetaSetMut("queue_storage_message_lag", msgLag)
-			}
-			for k, v := range metadata {
-				part.MetaSetMut(k, v)
+	n := int32(len(dequeue.Messages))
+	props, _ := queueClient.GetProperties(ctx, nil)
```

Comment:
I wonder if it's safe to swallow this error (even though the original code did that as well)